lib/graphql/graphql-client.ts

```diff
@@ -0,0 +1,40 @@
+import axios, { AxiosRequestConfig, AxiosResponse } from 'axios'
+
+import { GraphQLResponse } from './graphql-schemas'
+
+/* Interface for accessing any GraphQL API */
+export interface IGraphQLClient {
+  fetchData<T>(query: string, variables?: { [key: string]: any }): Promise<T>
+}
+
+/* Implementation of the IGraphQLClient interface to give access to any GraphQL API */
+export class GraphQLClient implements IGraphQLClient {
+  constructor(private readonly endpoint: string, private readonly headers: Record<string, string>) {}
+
+  async fetchData<T>(query: string, variables: { [key: string]: any } = {}): Promise<T> {
+    const requestConfig: AxiosRequestConfig = {
+      method: 'POST',
+      url: this.endpoint,
+      headers: this.headers,
+      data: { query, variables },
+    }
+
+    let response: AxiosResponse<GraphQLResponse<T>>
+    try {
+      response = await axios.request(requestConfig)
+    } catch (error) {
+      if (axios.isAxiosError(error)) {
```

Comment:
nit: we probably want to add both GQL call success and failure metrics, so that we can measure the GQL call success rate from routing-api perspective.

Reply:
we already log success/failure metrics in `graphql-token-fee-fetcher.ts` level, do you think we need extra logging here?

Reply:
sorry didn't see, nah we don't need.